shuffle/integration-tests/src/helper.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use diem_sdk::{client::BlockingClient, types::LocalAccount};
+use shuffle::{account, deploy, new, shared::Home};
+use std::{path::PathBuf, str::FromStr};
+use url::Url;
+
+pub struct ShuffleTestHelper {
+    home: Home,
+    tmp_dir: PathBuf,
+}
+
+impl ShuffleTestHelper {
+    pub fn new() -> Self {
+        let tmp_dir = std::env::temp_dir();
+        let home = Home::new(&tmp_dir).expect("unable to create home shuffle folder");
+        Self { tmp_dir, home }
+    }
+
+    pub fn home(&self) -> &Home {
+        &self.home
+    }
+
+    pub fn project_path(&self) -> PathBuf {
+        self.tmp_dir.join("project")
+    }
+
+    pub fn create_accounts(
+        &self,
+        treasury_account: &mut LocalAccount,
+        client: BlockingClient,
+    ) -> Result<()> {
+        account::create_local_accounts(&self.home, client, treasury_account)
+    }
+
+    pub fn create_project(&self) -> Result<()> {
+        new::handle(new::DEFAULT_BLOCKCHAIN.to_string(), self.project_path())
+    }
+
+    pub async fn deploy_project(&self, dev_api_url: &str) -> Result<()> {
+        let url = Url::from_str(dev_api_url)?;
+        deploy::handle(&self.home, &self.project_path(), url).await
+    }
+}
+
+impl Drop for ShuffleTestHelper {
+    fn drop(&mut self) {
+        // Clean up folder once instance is out of scope.
+        let _ = std::fs::remove_dir_all(&self.tmp_dir);
```

Comment:
This pattern of folder clean up when out of scope was picked up from `TempDir`: https://docs.rs/tempdir/0.3.5/src/tempdir/lib.rs.html#324

Reply:
Why don't you just use that directly instead of doing this yourslef?

Reply:
I had the mistaken belief that an instance of TempDir inside `pub fn new() -> Self {` would go out of scope and get dropped even if it was a field on the underlying struct. I've since gotten it working with TempDir and will push the change.